bundler/src/bundle.rs

```diff
@@ -185,3 +213,60 @@ where
         ])
     }
 }
+
+/// Read static data from files that should be included in the compiled bundle
+async fn static_data(patterns: &[Pattern]) -> Result<HashMap<String, Vec<u8>>, std::io::Error> {
+    let mut data = HashMap::new();
+    for pattern in patterns {
+        for file in glob::glob(pattern.as_str()).expect("Pattern was validated by CLI") {
+            let file = file.map_err(|e| e.into_error())?;
+            let name = file.file_stem();
+            let Some(name) = name.and_then(OsStr::to_str) else {
+                // Save having to think about non-utf8 in OPA rules
+                trace!("Skipping non-utf8 static file: {name:?}");
+                continue;
+            };
+            data.insert(name.to_string(), tokio::fs::read(&file).await?);
+        }
+    }
+    Ok(data)
+}
+
+/// Combination of possible errors when fetching data to create bundle
+#[derive(Debug)]
+pub enum BundleDataError {
+    /// Error fetching data from database
+    Sql(sqlx::Error),
+    /// Error fetching data from file
+    Static(std::io::Error),
+}
+
+impl From<sqlx::Error> for BundleDataError {
+    fn from(value: sqlx::Error) -> Self {
+        Self::Sql(value)
+    }
+}
+
+impl From<std::io::Error> for BundleDataError {
+    fn from(value: std::io::Error) -> Self {
+        Self::Static(value)
+    }
+}
+
+impl Display for BundleDataError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            BundleDataError::Sql(e) => write!(f, "Error reading dynamic data: {e}"),
+            BundleDataError::Static(e) => write!(f, "Error reading static data: {e}"),
+        }
+    }
+}
+
+impl std::error::Error for BundleDataError {
+    fn source(&self) -> Option<&(dyn std::error::Error + 'static)> {
+        match self {
+            BundleDataError::Sql(e) => Some(e),
+            BundleDataError::Static(e) => Some(e),
+        }
+    }
+}
```

Comment:
Maybe make use of `thiserror` here :stuck_out_tongue: 

```suggestion
/// Combination of possible errors when fetching data to create bundle
#[derive(Debug, thiserror::Error)]
pub enum BundleDataError {
    /// Error fetching data from database
    #[error("Error reading dynamic data: {0}")]
    Sql(#[from] sqlx::Error),
    /// Error fetching data from file
    #[error("Error reading static data: {0}")]
    Static(#[from] std::io::Error),
}
```

Reply:
Don't mind either way. I erred on the side of not bringing in the extra dependency.

Reply:
Ah. Thought it was already in here. Probably aught to upgrade my error handing from `anyhow` to `thiserror` at some point

Reply:
It is now